crowbar_framework/app/helpers/installer/upgrades_helper.rb

```diff
@@ -0,0 +1,79 @@
+
+#
+# Copyright 2011-2013, Dell
+# Copyright 2013-2015, SUSE Linux GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Installer
+  module UpgradesHelper
+    def continue_button(disabled = false)
+      if disabled
+        button_tag icon_tag(:chevron_right, t(".continue")), class: "btn btn-primary disabled"
+      else
+        button_tag icon_tag(:chevron_right, t(".continue")), class: "btn btn-primary"
+      end
+    end
+
+    def check_repos_button
+      if check_repos?
+        continue_button
+      else
+        button_tag icon_tag(:refresh, t(".recheck")), class: "btn btn-primary"
+      end
+    end
+
+    def alert_type(boolean)
+      if boolean
+        "alert-success"
+      else
+        "alert-danger"
+      end
+    end
+
+    def check_ha_repo?
+      return nil unless Proposal.where(barclamp: "pacemaker")
```

Comment:
```Proposal.where(barclamp: "pacemaker")``` always returns something that evaluates to  ```true``` you might try with:

```Proposal.where(barclamp: "pacemaker").first``` that will evaluate to ```nil``` in case there is no proposal.